lib/ASTGen/Sources/ASTGen/Diagnostics.swift

```diff
@@ -91,10 +91,18 @@ func emitDiagnostic(
   diagnostic: Diagnostic,
   messageSuffix: String? = nil
 ) {
+  // Determine the set of note IDs.
+  let knownNoteIDs: Set<MessageID> = .init(
+    diagnostic.notes.map { $0.noteMessage.fixItID }
+  )
```

Comment:
The Swift parser models diagnostics a little differently. Each diagnostic has both notes + fix-it's. A note has no fix-it attached (I think `fixItID` is actually misnamed here, it's just the note ID). A fix-it has both a message (or "note" if you like :P) and a set of changes.

We can talk about whether that's how we want to model that or not later, but IMO the way these should be emitted right now is that each of the fix-it's just become a note with an attached fix-it.

Reply:
This doesn't match the model used by the compiler, and is somewhat lossy. But I'll embrace the model here (see my latest commit) and we can iterate on the swift-syntax diagnostic model more.